tests/MongoDB.Driver.Examples/TransactionExamplesForDocs/WithTransactionExample1.cs

```diff
@@ -0,0 +1,100 @@
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Examples.TransactionExamplesForDocs
+{
+    public class WithTransactionExample1
+    {
+        [Fact]
+        public void Example1()
+        {
+            RequireServer.Check().ClusterTypes(ClusterType.ReplicaSet, ClusterType.Sharded).Supports(Feature.Transactions);
+
+            var connectionString = CoreTestConfiguration.ConnectionString.ToString();
+            DropCollections(
+                connectionString, 
+                CollectionNamespace.FromFullName("mydb1.foo"), 
+                CollectionNamespace.FromFullName("mydb2.bar"));
+            var result = string.Empty;
+
+            // Start Transactions withTxn API Example 1
+            // For a replica set, include the replica set name and a seedlist of the members in the URI string; e.g.
+            // string uri = "mongodb://mongodb0.example.com:27017,mongodb1.example.com:27017/?replicaSet=myRepl";
+            // For a sharded cluster, connect to the mongos instances; e.g.
+            // string uri = "mongodb://mongos0.example.com:27017,mongos1.example.com:27017:27017/";
+            var client = new MongoClient(connectionString);
+
+            // Prereq: Create collections. CRUD operations in transactions must be on existing collections.
+            var database1 = client.GetDatabase("mydb1");
+            var collection1 = database1.GetCollection<BsonDocument>("foo").WithWriteConcern(WriteConcern.WMajority);
+            collection1.InsertOne(new BsonDocument("abc", 0));
+
+            var database2 = client.GetDatabase("mydb2");
+            var collection2 = database2.GetCollection<BsonDocument>("bar").WithWriteConcern(WriteConcern.WMajority);
+            collection2.InsertOne(new BsonDocument("xyz", 0));
+
+            // Step 1: Start a client session.
+            using (var clientSession = client.StartSession())
+            {
+                // Step 2: Optional. Define options to use for the transaction.
+                var transactionOptions = new TransactionOptions(
+                    readPreference: ReadPreference.Primary,
+                    readConcern: ReadConcern.Local,
+                    writeConcern: WriteConcern.WMajority);
+
+                // Step 3: Define the sequence of operations to perform inside the transactions
+                result = clientSession.WithTransaction(
+                    (handle, token) =>
```

Comment:
When I first read this I didn't know what `token` was...

Rename to `cancellationToken`.

Reply:
First parameter would normally be named `session` but that conflicts with the existing session variable already in scope.

Rename to `s` or `_`?

Reply:
Done